src/git/parsers/blameParser.ts

```diff
@@ -69,7 +70,8 @@ export class GitBlameParser {
                             .slice(1)
                             .join(' ')
                             .trim();
-                        if (currentUser !== undefined && currentUser === entry.author) {
+                        if (currentUser !== undefined && currentUser === entry.author &&
```

Comment:
Rather than have these checks here, I think they should be moved to be at the end of the entry parsing -- probably in `parseEntry` https://github.com/eamodio/vscode-gitlens/blob/develop/src/git/parsers/blameParser.ts#L157

That way we don't need to have the check in 2 places, since we don't want to rely on the email being parsed before the author (which should actually be the other way around given the current structure)

Reply:
This applies to the other parsers as well